gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -3649,8 +3652,28 @@ protected final void releaseAfterGetInitialImageLatch() {
           + getFullPath());
     }
     releaseLatch(this.initializationLatchAfterGetInitialImage);
+
+    writeUnlockEnqueueDelta();
```

Comment:
How are we sure that this route has always acquired the lock before? And shouldn't this be in some finally block by the caller rather than being here.